### ISSUE TITLE
Long LDAP DN Cyrilic Users Login error when save file 

### ISSUE THREAD

**Issue by martingrigorev** [minio/minio#20797]
## NOTE
Hi ! First thank you for your product! Amazing work.

We are trying to add ldap authentication in latest minio installation
So, some of our accounts have a short DN and it's ok, But some of our 
workers are inside long-named department. And Active directory gives us 
Long dn decoded for cyrilic symbols. Minio writes a long named file on a File System 
Thats have a 256 bit limits. And users with Long DN just cant login due to this issue.

minio.server.com:9000  [STORAGE storage.ReadXL] [2024-12-27T15:10:36.730] /mnt/disk3 .minio.sys config/iam/policydb/sts-users/cn=\d0\9c\d0\b8\d1\88\d0\ .............\d0\b2\d0\b8\d1\87,ou=\d0\9e\d1\82\d0\b4\d0\b5\d0\bb \d0\b0\d0\b4\d0\bc\d0\b8\d0\bd\d0\b8\d1\81\d1\82\d1\80\d0\b8\d1\80\d0\be\d0\b2\d0\b0\d0\bd\d0\b8\d1\8f Unix-\d1\.................1\82\d0\b5\d0\bc,ou=\d0\a3\d0\bf\d1\80\d0\b0\d0\b2\d0\bb\d0\b5\d0\bd\d0\b8\d0\b5 \d0\b8\d0\bd\d1\84\d1\80\d0\b0\d1\81\d1\82\d1\80\d1\83\d0\ba\d1\82\d1\83\d1\80\d1\8b,ou=\d0\.............. \be\d0\bb\d0\be\........... b8\d0\b9,ou=mf,ou=uk,ou=Users,ou=department,dc=example,dc=com.json total-errs-availability=0 total-errs-timeout=0 err='file name too long'

We have got a error - File name too long.

## Expected Behavior

Expected to user with different lenghts in DN should login correctly on any language.

## Current Behavior

Users with long DN with coded Cyrilic symbols inside cant login

## Possible Solution

Do not write long json on file system or use another method.

## Steps to Reproduce (for bugs)

1. Centos fresh instal
2. Latest Minio
3. LDAP with cyrilic symbols 
4. Users with long DN

## Your Environment

* Version used (`minio --version`):
* Server setup and configuration:
* Operating System and version (`uname -a`):

**Comment by harshavardhana** [comment]
Yeah it's unfortunate we have no plans to extend our implementation to long Cyrillic names.

It is simply a very niche requirement for us.

**Comment by harshavardhana** [comment]
We may get to it if there is a large commercial requirement, for now it's a very low priority item for us.

**Comment by martingrigorev** [comment]
Maybe there's some workaround or something that will work in that situation. I think that problem actual not only for cyrilic countries users, it' s for any language that uses multibyte encoding in MS AD. Maybe you will add some variable that can solve this problem?

**Comment by klauspost** [comment]
It should store `Hex(SHA256("sts-users/cn=\d0\9c...."))`. But it would require a significant rewrite and testing, which I don't see happening soon.

**Comment by harshavardhana** [comment]
> It should store `Hex(SHA256("sts-users/cn=\d0\9c...."))`. But it would require a significant rewrite and testing, which I don't see happening soon.

Actually @klauspost Cyrillic's are handled by filesystem perfectly fine. The problem is we are trying to store an encoded value whereas we must not do that instead just save what is the actual "name" it will be fine.

Other than Windows as backend this is fine for almost all of Linux based deployments.

**Comment by harshavardhana** [comment]
Doing it hex would make it even more platform independent.

**Comment by harshavardhana** [comment]
> Maybe there's some workaround or something that will work in that situation. I think that problem actual not only for cyrilic countries users, it' s for any language that uses multibyte encoding in MS AD. Maybe you will add some variable that can solve this problem?

Yeah multi-byte encoding is the problem however it is not for all lengths it is for characters that exceed in length by 256 characters per path component. This happens usually for unexpected lengths such as Cyrillic characters. 

It can be resolved via doing a sha256 and hex encoding out of it as @klauspost mentioned it's a full rewrite and not something which is a priority right now.

**Comment by klauspost** [comment]
@harshavardhana The problem is the length, and this seems to fail on linux.  Using SHA256 for these gives a fixed length and should be strong enough for it to never collide (intentionally or unintentionally).

**Comment by harshavardhana** [comment]
> @harshavardhana The problem is the length, and this seems to fail on linux. Using SHA256 for these gives a fixed length and should be strong enough for it to never collide (intentionally or unintentionally).

Yeah that is safer, and we should have done this before.

**Comment by nicholasgribanov** [comment]
Hello! Something new about issue ? We have the same problem

**Comment by klauspost** [comment]
We will not be addressing this in the Community Edition.
